### ISSUE TITLE
Metadata schema for images and documents

### ISSUE THREAD

**Issue by geoffroy-noel-ddh** [kingsdigitallab/django-kdl-wagtail#8]
In the Field project they need some basic metadata for papers (wagtail documents) and photographs, such as date, authors, ...

I'm thinking of generalising that into a Dublin Core schema/table which could have a one-to-one relationship with Image and Doc. I haven't yet thought exactly how it would work but the idea is have something reusable and that can be applied to existing records (even other than Image or Document) for search and rendering purpose.

I'll test the idea first with Field then try to move it to a non-core app in kdl-wagtail if there is interest in that. Feel free to make suggestions.

**Comment by jmiguelv** [comment]
Would it make sense to actually have custom Image and Document models that have support for the DC fields by default? They would not need to be required, but at least it would enforce some good practice.

**Comment by geoffroy-noel-ddh** [comment]
Yes, I don't see why not. I don't have a full plan for this yet but I think having a separate table for DC that could link to any content type would keep things generic (we could then apply it to other Django models if needed and index everything in the same way). Although I'm not a fan of Django generic relationships. Also there is the question of where would DB fields be editable in the backend interface. Ideally on the image and doc themselves but presumably in a separate admin as well (so we can search and edit across content types).

**Comment by geoffroy-noel-ddh** [comment]
I was also thinking of maybe using DC to store bibliographical records, which is a common requirements in our projects. But I'm not sure DC is the best way to store that kind of information so it might need to be complemented with another model. @brian-maher I think you did a fairly generic bib records model for a project recently?
